sympy/solvers/ode/tests/test_ode.py

```diff
@@ -33,6 +33,112 @@
 # dsolve() to return different results on different machines)
 
 
+def _ode_solver_test(ode_examples,our_hint='default'):
+    for example in ode_examples:
+        eq = ode_examples[example]['eq']
+        sol = ode_examples[example]['sol']
+        assert our_hint in classify_ode(eq), our_hint+"Hint was not recognised for the equation "+str(eq)
+        sols = sol + [constant_renumber(root) for root in sol]
```

Comment:
Is it necessary to have `constant_renumber` here? I would think it better to renumber the examples manually.

Reply:
so how should we handle this if the order of terms is changed with constants?

Reply:
Can you give an example?

I think ideally `constantsimp` should ensure that the constants are always numbered in a canonical way.

Reply:
let us say that 
`sol = C3 + C2*x + C1*x**2` but solution returned by dsolve is `C1*x**2  + C2 + C3*x` so i guess it will give assertion error

Reply:
The `constant_renumber` function is called at the end of `dsolve` so ordering should be the same every time:
https://github.com/sympy/sympy/blob/5b92c4497fcc6f1df4aac23b9c001ff323ffb421/sympy/solvers/ode/ode.py#L2492

Reply:
so should I remove the extra solutions which are added with the use of `constant_renumber`?

Reply:
Yes and if necessary change the solutions that are in the test examples structure